crates/polars-plan/src/dsl/function_expr/strings.rs

```diff
@@ -405,11 +405,20 @@ pub(super) fn extract_all(args: &[Series]) -> PolarsResult<Series> {
     }
 }
 
-pub(super) fn count_match(s: &Series, pat: &str) -> PolarsResult<Series> {
-    let pat = pat.to_string();
+pub(super) fn count_match(args: &[Series]) -> PolarsResult<Series> {
+    let s = &args[0];
+    let pat = &args[1];
 
     let ca = s.utf8()?;
-    ca.count_match(&pat).map(|ca| ca.into_series())
+    let pat = pat.utf8()?;
+    if pat.len() == 1 {
+        let pat = pat
+            .get(0)
+            .ok_or_else(|| polars_err!(ComputeError: "expected a pattern, got null"))?;
```

Comment:
I'm worried that this error will break generality. If it is OK to have an entire column of `null` patterns, it should also be OK to have a column of length 1 of patterns that is `null`. I think we should return a full-null series of length `s` in this case.

Reply:
Right now this is exactly the same as `extract_all`. Maybe it's a good idea to align all string-regex methods that accept `Expr`. I'll await your reply.